lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -29,22 +29,44 @@ def test_get_bad_keys(self, client, server_config, attach_dataset):
         )
         assert response.status_code == HTTPStatus.BAD_REQUEST
         assert response.json == {
-            "message": "Unrecognized list values ['xyzzy', 'plugh'] given for parameter metadata; expected saved,seen,user,deletion,access,owner"
+            "message": "Unrecognized list values ['plugh', 'xyzzy'] given for parameter metadata; expected dashboard.saved,dashboard.seen,user.,server.deletion,access,owner"
         }
 
-    def test_get(self, client, server_config, provide_metadata):
+    def test_get1(self, client, server_config, provide_metadata):
         response = client.get(
             f"{server_config.rest_uri}/datasets/metadata",
             query_string={
                 "controller": "node",
                 "name": "drb",
-                "metadata": ["seen", "saved", "access", "user"],
+                "metadata": [
+                    "dashboard.seen",
+                    "dashboard.saved",
+                    "access",
+                    "user.contact",
```

Comment:
But `access` and `owner` aren't in the `server` Metadata namespace: they're primary `Dataset` columns. While I could easily fake a separate namespace, (maybe `dataset.access`), I saw no need for such obfuscation. They're special; but for the client it's just another keyword.

Reply:
As I mentioned [above](https://github.com/distributed-system-analysis/pbench/pull/2417#discussion_r706260337), I think we should be careful about letting the database representation drive or otherwise show through the interface.

Indeed, from the Server's  internals perspective, `"access"` _is_ a primary `Dataset` column, but from the client perspective, via the API, `"access"` is a metadata field, like the others in that code snippet, and, as such, it's weird that it doesn't have a namespace prefix.

Giving it a namespace prefix wouldn't prevent the Server code from resolving it to a `Dataset` column (it already manages to do that now), but it would present a uniform approach as part of the interface, which hopefully would make things easier to understand and support.

Reply:
In a way, though, what your suggesting is that the keyword namespace *should be* tied to the implementation, because otherwise it makes no difference what the "access" keyword is.

Still, I've already changed this in conjunction with your nudge to create constants for the "access" and "owner" keywords; although I've tied them specifically to the Dataset, as `dataset.access` and `dataset.owner`.

Reply:
> what you're suggesting is that the keyword namespace should be tied to the implementation

No, I'm trying to impose a small amount of architecture or interface definition; what the implementation of that is is fairly wide open.

But, yes, I'm asserting that the `access` value "belongs" to the `server` rather than the client (`dashboard`) and that, as a piece of metadata, it does not occupy a root-level position (I would prefer to reserve those nodes for namespace subtrees).

It sounds like you're introducing a new root-level subtree.  I'll be interested to see what that's about and whether/how it helps.  (E.g., as I understand it, _all_ the metadata items that we're discussing are tied to a given dataset....)